tests/lib/c_lib/src/main.c

```diff
@@ -319,6 +322,242 @@ void test_bsearch(void)
 	zassert_not_null(result, "bsearch -key found");
 }
 
+/**
+ *
+ * @brief Test abs function
+ *
+ */
+
+void test_abs(void)
+{
+	int val = -5, value = 5;
+
+	zassert_equal(abs(val), 5, "abs -5");
+	zassert_equal(abs(value), 5, "abs 5");
+}
+
+/**
+ *
+ * @brief Test atoi function
+ *
+ */
+void test_atoi(void)
+{
+	zassert_equal(atoi("123"), 123, "atoi error");
+	zassert_equal(atoi("2c5"), 2, "atoi error");
+	zassert_equal(atoi("acd"), 0, "atoi error");
+	zassert_equal(atoi(" "), 0, "atoi error");
+	zassert_equal(atoi(""), 0, "atoi error");
+	zassert_equal(atoi("3-4e"), 3, "atoi error");
+	zassert_equal(atoi("8+1c"), 8, "atoi error");
+}
+
+/**
+ *
+ * @brief Test value type
+ *
+ * @details This function check the char type,
+ * and verify the return value.
+ *
+ * @see isalnum(), isalpha(), isdigit(), isgraph(),
+ * isprint(), isspace(), isupper(), isxdigit().
+ *
+ */
+void test_checktype(void)
+{
+	static const char exp_alnum[] =
+		"0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+	static const char exp_alpha[] =
+		"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+	static const char exp_digit[] = "0123456789";
+	static const char exp_graph[] =
+		"!\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
+	static const char exp_print[] =
+		" !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
+	static const char exp_space[] = {"\x9\xa\xb\xc\xd\x20"};
+
+	static const char exp_upper[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+	static const char exp_xdigit[] = "0123456789ABCDEFabcdef";
+	char buf[128];
+	char *ptr = buf;
+
+	for (int i = 0; i < 128; i++) {
+		if (isalnum(i))
+			*ptr++ = i;
```

Comment:
IIRC, CI used to give warnings for these. I wonder what happened to that ... cc @nashif @galak